### DISCUSSION TITLE
Convert from multivariate polynomial to univariate

### DISCUSSION THREAD

**Discussion by adhalanay** [oscar-system/Oscar.jl#3772] [Q&A] [answered]
Assume that one defined a polynomial ring `R,(x,y,z)=polynomial_ring(QQ,["x","y","z"])` and an element $f$ of $R$. Define `R0,(x,y)=polynomial_ring(QQ,["x","y"]` can one convert $f$ to an univariate polynomial from the ring `R1,z=polynomial_ring(R0,"z")`?

**Edit:** Defining `R0` ,then `R1` and `f=x*y*z` fails with 

> ERROR: Cannot promote to common type

**Comment by joschmitt** [comment, marked as answer]
You can 'evaluate' the multivariate polynomial at the new variables:
```
julia> R, (x, y, z) = polynomial_ring(QQ, ["x", "y", "z"]);

julia> R0, (u, v) = polynomial_ring(QQ, ["u", "v"]);

julia> R1, w = polynomial_ring(R0, "w");

julia> f = x*y*z # This lives in R
x*y*z

julia> f(u, v, w) # This now lives in R1
u*v*w
```
In the same way, you can build the natural homomorphism from `R` to `R1`:
```
julia> phi = hom(R, R1, [u, v, w])
Ring homomorphism
  from multivariate polynomial ring in 3 variables over QQ
  to univariate polynomial ring in w over R0
defined by
  x -> u
  y -> v
  z -> w

julia> phi(f)
u*v*w
```

I cannot reproduce the error you describe. Can you share the complete code?

**Reply by adhalanay** [reply]
Indeed, there was a silly mistake in my code. I had written `R1,z=polynomial_ring(QQ,"z")`
